code/modules/reagents/reagent_containers/cups/drinks.dm

```diff
@@ -443,6 +450,57 @@
 		desc += " It has an emblazoned Nanotrasen logo on it."
 		icon_state = "shaker_n"
 
+/obj/item/reagent_containers/cup/glass/shaker/examine(mob/user)
+	. = ..()
+	. += span_notice("Alt-click to [using_custom_drinks?"disable":"enable"] custom drink naming")
+	if(using_custom_drinks)
+		. += span_notice("Drinks poured from this shaker will have the following name: [custom_drink_name]")
+		. += span_notice("Drinks poured from this shaker will have the following description: [custom_drink_desc]")
+
+/obj/item/reagent_containers/cup/glass/shaker/AltClick(mob/user)
+	. = ..()
+	using_custom_drinks = !using_custom_drinks
+
+	if(!using_custom_drinks)
+		disable_custom_drinks()
+		balloon_alert(user, "custom drinks disabled!")
+		return
+
+	var/new_name = reject_bad_text(tgui_input_text(user, "Drink name", "Set drink name", custom_drink_name, 45, FALSE), 64)
+	if(!new_name)
+		balloon_alert(user, "invalid drink name!")
+		using_custom_drinks = FALSE
+		return
+
+	var/new_desc = reject_bad_text(tgui_input_text(user, "Drink description", "Set drink description", custom_drink_desc, 64, TRUE), 128)
+	if(!new_desc)
+		balloon_alert(user, "invalid drink description!")
+		using_custom_drinks = FALSE
+		return
```

Comment:
These inputs could also use sanity checks that the user is still in a valid state to do this after closing the text box. 

Otherwise you could, say, alt-click all the bartender's shakers and hold the input box open, go across the station, and then rename all of their drinks while they're in the middle of pouring.